test/unit/math/prim/arr/fun/append_test.cpp

```diff
@@ -0,0 +1,43 @@
+#include <stan/math/prim/arr.hpp>
+#include <gtest/gtest.h>
+
+TEST(MathFunctions, append_array) {
```

Comment:
You generally want to test boundary conditions, like one or both of the vectors being size zero.  I know the rest of our tests haven't been that thorough.